examples/md-flexible/parsing/CLIParser.cpp

```diff
@@ -526,7 +535,7 @@ bool checkFileExists(const std::string &filename) {
 }  // namespace
 
 void CLIParser::inputFilesPresent(int argc, char **argv, MDFlexConfig &config) {
-  int option, optionIndex;
+  int option = 0, optionIndex = 0;
```

Comment:
You might want to make the while loop below into a for loop and move the current `option` and `optionIndex` into its initializer statement.
You might also want to rename the variable `option`, as currently it just creates confusion between the class `option` and the variable `option`.

Reply:
And how do you propose this for loop header should look like?
```C++
  for (int cliOptionIndex = 0, cliOption = cliOption = getopt_long(argc, argv, "", long_options, &cliOptionIndex);
       cliOption != -1; cliOption = cliOption = getopt_long(argc, argv, "", long_options, &cliOptionIndex)) {
```
this does not look very appealing and just creates code duplication and more confusion :/

Reply:
```c++
for (int option = 0, optionIndex = 0; (option = getopt_long(argc, argv, "", longOptions, &optionIndex)) != -1; ) {
```

Reply:
just to have things in the proper scope.

Reply:
ugly but I like it.

Reply:
done in 7825fc3